tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -114,3 +119,48 @@ def enhance_policies(self, access_policies):
             policy['principalName'] = GraphHelper.get_principal_name(aad_object)
 
         return access_policies
+
+
+@resources.register('key')
+class KeyVaultKey(ArmResourceManager):
+
+    class resource_type(ArmResourceManager.resource_type):
+        service = 'azure.mgmt.keyvault'
+        client = 'KeyVaultManagementClient'
+        enum_spec = ('vaults', 'list', None)
+
+    def augment(self, resources):
+        url_base = 'vault.azure.net'
+        s = Session(resource='https://' + url_base)
+        self.keyvault_client = KeyVaultClient(s.get_credentials())
+        r = []
+        for index, item in enumerate(resources):
+            vault_url = 'https://' + resources[index]["name"] + '.vault.azure.net'
+            try:
+                vkeys = self.keyvault_client.get_keys(vault_url)
+                for vkey in vkeys:
+                    r.append({'kid': vkey.kid,
+                              'keyvault': {'id': resources[index]["id"],
+                                           'name': resources[index]["name"],
+                                           'location': resources[index]["location"],
+                                           'tags': resources[index]["tags"]},
+                              'enabled': vkey.attributes.enabled,
+                              'created': vkey.attributes.created,
+                              'updated': vkey.attributes.updated,
+                              'tags': vkey.tags})
+
+            except Exception as e:
+                self.log.error(e.message + ' on keyvault: ' + resources[index]["id"])
+                pass
```

Comment:
hmm.. why should this pass instead of raising?

Reply:
Because I may not have permissions to a storage account that's in my current subscription and I will get auth access errors...similar to s3 auth errors if I don't have permissions as I loop through them. I think it's perfectly fine and ok if you don't have access to a storage account so logging the error message with a pass is ok. discussed with @thisisshi  and he was ok with how I wrote it. Let me know if this is a show stopper to getting this pull approved once I have my final commits in and I can comment further.

Reply:
rather than doing the try/except on a generic Exception we should be catching on specific ones here.

Reply:
at minimum this try/except should be just around the api call, and catch a specific api client exception, not generic Exception. if an error is persistent wrt to access, i tend to think the policy should fail, unless its explicit about the type of error its catching (ie permission denied) and ideally that behavior is documented. in general we don't catch permission denied, because it means the policy will never work without human intervention, and the  best way to get that is for the policy to error out (error metrics, etc).